lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -278,10 +286,15 @@ def grafana_create_dashboard(module, data):
                 result['changed'] = True
             else:
                 body = json.loads(info['body'])
-                raise GrafanaAPIException('Unable to update the dashboard %s : %s' % (uid, body['message']))
+                raise GrafanaAPIException(
+                    'Unable to update the dashboard %s : %s' % (uid, body['message']))
     else:
         # create
-        r, info = fetch_url(module, '%s/api/dashboards/db' % data['grafana_url'], data=json.dumps(payload), headers=headers, method='POST')
+        if 'id' in payload["dashboard"]:
```

Comment:
Only those two lines are required, and it works for me with it

Reply:
@nerzhul How I can check out your code / changes - I will test this very quickly with test case which I pasted already above?

Reply:
can you explain the purpose of you change ?

Reply:
Hi @mullnerz, @jorijinnall - I've created quick code / place to test this. Some improvements can be done but this is for start point. Allows quickly test ansible / code with fix whether allows to create / and update dashboard: Feel free to update code to make these tests faster - so maybe we will add this ansible tests later.
- https://github.com/marcinpraczko/ansible-simple-testing/tree/master/ansible-issue-46152